contrib/platform/src/com/sun/jna/platform/win32/Crypt32.java

```diff
@@ -142,4 +145,203 @@ public boolean CryptUnprotectData(DATA_BLOB pDataIn, PointerByReference szDataDe
+	 *            current state of the root store and not the state of the root
+	 *            store at a time passed in by this parameter. For revocation, a
+	 *            certificate revocation list (CRL), itself, must be valid at the
+	 *            current time. The value of this parameter is used to determine
+	 *            whether a certificate listed in a CRL has been revoked.
+	 * @param hAdditionalStore
+	 *            A handle to any additional store to search for supporting
+	 *            certificates and certificate trust lists (CTLs). This parameter
+	 *            can be NULL if no additional store is to be searched.
+	 * @param pChainPara
+	 *            A pointer to a CERT_CHAIN_PARA structure that includes
+	 *            chain-building parameters.
+	 * @param dwFlags
+	 *            Flag values that indicate special processing. This parameter can
+	 *            be a combination of one or more of the following flags.
+	 * @param pvReserved
+	 *            This parameter is reserved and must be NULL.
+	 * @param ppChainContext
+	 *            The address of a pointer to the chain context created. When you
+	 *            have finished using the chain context, release the chain by
+	 *            calling the CertFreeCertificateChain function.
+	 * @return If the function succeeds, the function returns nonzero (TRUE). If the
+	 *         function fails, it returns zero (FALSE).
+	 */
+	boolean CertGetCertificateChain(Pointer hChainEngine, CERT_CONTEXT pCertContext, Pointer pTime,
+			Pointer hAdditionalStore, CERT_CHAIN_PARA pChainPara, int dwFlags, Pointer pvReserved,
+			CERT_CHAIN_CONTEXT ppChainContext);
+
+	/**
+	 * The CertFreeCertificateContext function frees a certificate context by
+	 * decrementing its reference count. When the reference count goes to zero,
+	 * CertFreeCertificateContext frees the memory used by a certificate context.
+	 *
+	 * @param pCertContext
+	 *            A pointer to the CERT_CONTEXT to be freed.
+	 * @return The function always returns nonzero.
+	 */
+	boolean CertFreeCertificateContext(CERT_CONTEXT pCertContext);
+
+	/**
+	 * The CertCloseStore function closes a certificate store handle and reduces the
+	 * reference count on the store. There needs to be a corresponding call to
+	 * CertCloseStore for each successful call to the CertOpenStore or
+	 * CertDuplicateStore functions.
+	 *
+	 * @param hCertStore
+	 *            Handle of the certificate store to be closed.
+	 * @param dwFlags
+	 *            Typically, this parameter uses the default value zero. The default
+	 *            is to close the store with memory remaining allocated for contexts
+	 *            that have not been freed. In this case, no check is made to
+	 *            determine whether memory for contexts remains allocated.
+	 * @return If the function succeeds, the return value is TRUE. If the function
+	 *         fails, the return value is FALSE.
+	 */
+	boolean CertCloseStore(HANDLE hCertStore, int dwFlags);
```

Comment:
See comment for `CertGetCertificateChain` the `hCertStore` parameter is used inconsistently. It would be good if this would be unified.